thrust/detail/config/compiler.h

```diff
@@ -71,10 +71,10 @@
 #define THRUST_DEVICE_COMPILER THRUST_DEVICE_COMPILER_GCC
 #elif THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_CLANG
 // CUDA-capable clang should behave similar to NVCC.
-#if defined(__CUDA__)
-#define THRUST_DEVICE_COMPILER THRUST_DEVICE_COMPILER_NVCC
-#else
+#if defined(__CUDA__) && defined(__clang__)
```

Comment:
`#if defined __CUDACC__` on line 66 would still set compiler to THRUST_DEVICE_COMPILER_NVCC and we'll never make it here.